lib/resourcecontrol/resource_control.go

```diff
@@ -0,0 +1,205 @@
+package resourcecontrol
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"os/exec"
+	"path"
+	"strconv"
+
+	"golang.org/x/sys/unix"
+)
+
+// NOTE: calling log.Fatal() inside a library is usually considered bad
+// practice, in this case it only happens in the child process so the library
+// user will not be affected.
+
+const (
+	execEnvVar      = "OVERSEER_EXEC"
+	cpuMaxEnvVar    = "OVERSEER_CPU_MAX"
+	memMaxEnvVar    = "OVERSEER_MEM_MAX"
+	ioMaxRbpsEnvVar = "OVERSEER_IO_MAX_RBPS"
+	ioMaxWbpsEnvVar = "OVERSEER_IO_MAX_WBPS"
+
+	controlSubtree = "overseer"
+
+	errPipeFd = 3
+)
+
+var (
+	errCgroupNotMounted = errors.New("cgroup2 is not mounted")
+	errNotEnoughArgs    = errors.New("not enough arguments provided")
+)
+
+// TODO: add more limits
+type ResourceLimits struct {
+	CPUMax    string
+	MemMax    string
+	IOMaxRbps string
+	IOMaxWbps string
+}
+
+func writeAndDie(f *os.File, m error) {
+	if _, err := f.Write([]byte(m.Error())); err != nil {
+		log.Fatal(err)
+	}
+
+	os.Exit(1)
+}
+
+// init will look for a special environment variable that, if present, will
+// signal the need to set up the resource limits and call unix.Exec() to start
+// the limited command
+func init() {
+	if os.Getenv(execEnvVar) == execEnvVar {
```

Comment:
Nit: From readability perspective, it might make sense to invert this condition to return early and reduce indentation below.